spring-integration-file/src/main/java/org/springframework/integration/file/FileReadingMessageSource.java

```diff
@@ -173,21 +179,20 @@ public void setAutoCreateDirectory(boolean autoCreateDirectory) {
 	}
 
 	/**
-	 * Sets a {@link FileListFilter}. By default a
-	 * {@link org.springframework.integration.file.filters.AbstractFileListFilter}
+	 * Sets a {@link FileListFilter}.
+	 * By default a {@link org.springframework.integration.file.filters.AcceptOnceFileListFilter}
 	 * with no bounds is used. In most cases a customized {@link FileListFilter} will
-	 * be needed to deal with modification and duplication concerns. If multiple
-	 * filters are required a
+	 * be needed to deal with modification and duplication concerns.
+	 * If multiple filters are required a
 	 * {@link org.springframework.integration.file.filters.CompositeFileListFilter}
 	 * can be used to group them together.
 	 * <p>
 	 * <b>The supplied filter must be thread safe.</b>.
-	 *
 	 * @param filter a filter
 	 */
 	public void setFilter(FileListFilter<File> filter) {
 		Assert.notNull(filter, "'filter' must not be null");
-		this.scanner.setFilter(filter);
+		this.filter = filter;
 	}
```

Comment:
I think we'll need a `getScanner()` so the filter can still be changed after initialization.

We'll also need a migration guide entry to state that calling this will not change the filter after bean initialization.